File: src/filters/search_filter.py
class SearchFilter:
    """
    Search and filter criteria for journal entries.
    """
    def __init__(self, category: str = None, keyword: str = "", date: str = ""):
        self.__category = category
        self.__keyword = keyword
        self.__date = date

    def is_valid_filter(self) -> bool:
        """Checks if at least one parameter is filled."""
        return bool(self.__category or self.__keyword or self.__date)

    def apply_filter(self):
        """Logic placeholder for applying filter."""
        pass

    # Getters
    def get_category(self): return self.__category
    def get_keyword(self): return self.__keyword
    def get_date(self): return self.__date

File: src/main/java/com/juki/model/SearchFilter.java
package com.juki.model;

import java.time.LocalDate;
import java.time.LocalTime;

public class SearchFilter {
    private String category;
    private String keyword;
    private LocalDate date;

    public SearchFilter(String category, String keyword, LocalDate date) {
        this.category = category;
        this.keyword = keyword;
        this.date = date;
    }

    public boolean isValidFilter() {
        return (keyword != null && keyword.length() >= 3) || 
               (category != null && !category.isEmpty()) || 
               (date != null);
    }

    public void applyFilter() {
        if (this.keyword != null) {
            this.keyword = this.keyword.toLowerCase();
        }
    }

    public String getCategory() { return category; }
    public void setCategory(String category) { this.category = category; }

    public String getKeyword() { return keyword; }
    public void setKeyword(String keyword) { this.keyword = keyword; }

    public LocalDate getDate() { return date; }
    public void setDate(LocalDate date) { this.date = date; }
}
